Templates/AppSource App/.github/workflows/_BuildALGoProject.yaml

```diff
@@ -75,6 +75,7 @@ on:
 permissions:
   contents: read
   actions: read
+  id-token: write
```

Comment:
will test

Reply:
You are right that it failed because the reusable workflow cannot add more permissions than given by the caller.
I have added id-token to the caller - as federated credentials actually have a scope specifically for PR's:
![image](https://github.com/microsoft/AL-Go/assets/10775043/bdbf2ae7-103d-4f8e-8e36-3b1d97ee8462)

Meaning that the workflow doesn't have access to the secrets unless this scope is added.

The alternative would be to remove the permissions from _BuildALGoProject and specify them when calling the reuseable workflow, but then we would remove functionality from the pull request which is allowed by Azure KeyVault.